google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/flexible/AppEngineFlexibleDeploymentEditor.java

```diff
@@ -186,14 +194,22 @@ private void resetRuntimeDisplay() {
     }
   }
 
+  private void updateStagedArtifactNameEmptyText() {
+    Path artifactPath = Paths.get(Strings.nullToEmpty(deploymentSource.getFilePath()));
```

Comment:
this logic needs to be updated now doesn't it?

empty text should always be the source artifact name.